pkg/controller/controller.go

```diff
@@ -82,6 +110,40 @@ const (
 	snapshotAPIGroup = snapapi.GroupName // "snapshot.storage.k8s.io"
 )
 
+var (
```

Comment:
What about a function
```golang
generateCSIParams(actor, key string, deprecated bool)
```
and
```golang
noActor = ""
provisionerActor = "Provisioner"
controllerPublishActor = "ControllerPublish"
...
secretNameKey = "SecretName"
fsTypeKey = "FSType"
...
actors = []string{provisionerActor, controllerPublishActor, ...}
keys = []string{secretNameKey, fsTypeKey, ...}
```

Reply:
how would we associate actors with keys? This ends up just shifting the variable definition into a different type of struct I think. It's unclear what params we're trying to generate here, whats the output of this function?

Reply:
Sorry the func should be called `generateCSIParamKeys` and returns an actual storageclass parameter key. The logic would be the same as what you have right now, but encapsulated in a function, so it's easier to read and you don't need a giant list of variables.

Reply:
I'm trying to see if there's a way to not have a hard-coded list of all the various combinations.